File: GrowthWare_CS/GrowthWareFrameworkModel/Profiles/MNameValuePair.cs

using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using System.Web;
using GrowthWare.Framework.Enumerations;
using GrowthWare.Framework.ModelObjects.Base;

namespace GrowthWare.Framework.ModelObjects
{
	/// <summary>
	/// Base properties a name value pair
	/// </summary>
	/// <remarks>
	/// Corresponds to table ZFC_NVP and 
	/// Store procedures: 
	/// ZFP_SET_NVP, ZFP_GET_NVP
	/// </remarks>
	[Serializable(), CLSCompliant(true)]
	public class MNameValuePair : MProfile
	{

		#region "Member Properties"
			private string m_SchemaName = string.Empty;
			private string m_StaticName = "NEW";
			private string m_Display = string.Empty;
			private string m_Description = string.Empty;
			private int m_Status = -1;
		#endregion

		#region "Protected Methods"
		protected new void init(DataRow Datarow)
		{
			base.init(Datarow);
			base.setInt(ref base.m_ID, ref Datarow, "NVP_SEQ_ID");
			base.setString(ref m_SchemaName, ref Datarow, "SCHEMA_NAME");
			base.setString(ref m_StaticName, ref Datarow, "STATIC_NAME");
			base.m_Name = m_StaticName;
			base.setString(ref m_Display, ref Datarow, "DISPLAY");
			base.setString(ref m_Description, ref Datarow, "DESCRIPTION");
			base.setInt(ref m_Status, ref Datarow, "STATUS_SEQ_ID");
		}
		#endregion

		#region "Public Methods"
		/// <summary>
		/// Provides a new account profile with the default vaules
		/// </summary>
		/// <remarks></remarks>

		public MNameValuePair()
		{
		}

		/// <summary>
		/// Will populate values based on the contents of the data row.
		/// </summary>
		/// <param name="dr">Datarow containing base values</param>
		/// <remarks>
		/// Class should be inherited to extend to your project specific properties
		/// </remarks>
		public MNameValuePair(DataRow dr)
		{
			init(dr);
		}
		#endregion

		#region "Public Properties"
		public int Status {
			get { return m_Status; }
			set { m_Status = value; }
		}

		public string SchemaName
		{
			get{ return m_SchemaName; }
			set{ m_SchemaName = value.Trim(); }
		}

		public string StaticName {
			get { return m_StaticName; }
			set { m_StaticName = value.Trim(); }
		}

		public string Display {
			get { return m_Display; }
			set { m_Display = value.Trim(); }
		}

		public string Description {
			get { return m_Description; }
			set { m_Description = value.Trim(); }
		}
		#endregion
	}
}

File: GrowthWare_VB/GrowthWareFrameworkModel/Profiles/MNameValuePair.vb
Imports GrowthWare.Framework.Enumerations
Imports GrowthWare.Framework.ModelObjects.Base

Namespace ModelObjects
    ''' <summary>
    ''' Base properties a name value pair
    ''' </summary>
    ''' <remarks>
    ''' Corresponds to table ZFC_NVP and 
    ''' Store procedures: 
    ''' ZFP_SET_NVP, ZFP_GET_NVP
    ''' </remarks>
    <Serializable(), CLSCompliant(True)> _
    Public Class MNameValuePair
		Inherits MProfile

#Region "Member Properties"
        Private mStaticName As String = "NEW"
        Private mDisplay As String = String.Empty
        Private mDescription As String = String.Empty
        Private mStatus As Integer = -1
#End Region

#Region "Protected Methods"
        Public Shadows Sub Init(ByVal dr As DataRow)
            On Error Resume Next
            MyBase.Init(dr)
            MyBase.m_ID = CInt(dr("NVP_SEQ_ID"))
            mStaticName = CStr(dr("STATIC_NAME"))
            MyBase.m_Name = mStaticName
            mDisplay = CStr(dr("DISPLAY"))
            mDescription = CStr(dr("DESCRIPTION"))
            mStatus = CInt(dr("STATUS_SEQ_ID"))
        End Sub
#End Region

#Region "Public Methods"
        ''' <summary>
        ''' Provides a new account profile with the default vaules
        ''' </summary>
        ''' <remarks></remarks>
        Public Sub New()

        End Sub

        ''' <summary>
        ''' Will populate values based on the contents of the data row.
        ''' </summary>
        ''' <param name="dr">Datarow containing base values</param>
        ''' <remarks>
        ''' Class should be inherited to extend to your project specific properties
        ''' </remarks>
        Public Sub New(ByVal dr As DataRow)
            Init(dr)
        End Sub
#End Region

#Region "Public Properties"
        Public Property Status() As Integer
            Get
                Return mStatus
            End Get
            Set(ByVal value As Integer)
                mStatus = value
            End Set
        End Property

        Public Property StaticName() As String
            Get
                Return mStaticName
            End Get
            Set(ByVal Value As String)
                mStaticName = Value.Trim
            End Set
        End Property

        Public Property Display() As String
            Get
                Return mDisplay
            End Get
            Set(ByVal Value As String)
                mDisplay = Value.Trim
            End Set
        End Property

        Public Property Description() As String
            Get
                Return mDescription
            End Get
            Set(ByVal Value As String)
                mDescription = Value.Trim
            End Set
        End Property
#End Region
    End Class
End Namespace